### PR TITLE
New profiles for balsa,trojita,kube

### PR DESCRIPTION
**Pull request by kortewegdevries**

### CODE REVIEW

File: etc/profile-a-l/balsa.profile [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# Persistent local customizations
+include balsa.local
+# Persistent global definitions
+include globals.local
+
+noblacklist ${HOME}/.balsa
+# Comment to use GPG
+noblacklist ${HOME}/.gnupg
+noblacklist ${HOME}/mail
+noblacklist /var/spool/mail
```
Comment (rusty-snake) [inline etc/profile-a-l/balsa.profile:13]: - `noblacklist /var/mail`
- no `whitelist`?

File: etc/profile-a-l/kube.profile [hunk 2]
```diff
@@ -0,0 +1,64 @@
+# Firejail profile for kube
+# Description: Qt mail client
+# This file is overwritten after every install/update
+# Persistent local customizations
+include kube.local
+# Persistent global definitions
+include globals.local
+
+# Comment to use GPG
+noblacklist ${HOME}/.gnupg
```
Comment (rusty-snake) [inline etc/profile-a-l/kube.profile:10]: ???

**Review by rusty-snake — COMMENTED** [review]
LGTM

- IIRC, GPG needs `writable-run-user`
- @ all, do we want GPG opt-in or opt-out? 
- whitelisting-profiles would be better since these programs heavily deal with untrusted/internet-data.

**Comment by kortewegdevries** [comment]
1. Do they? As I checked they need the ~/.gnupg directory rw only.
2. My take is don't enable gpg by default.
3. I thought about making whitelisted...then you need to modify the existing like geary,evolution etc. add browser support+they create lot of extra folders for storing contacts,address book, in non-standard locations which I don't use...

**Comment by kortewegdevries** [comment]
I think we should make a allow-gpg/ssh.inc to make things shorter in profiles, It would have noblacklist/whitelist, /usr/share/gpg* ~/.gnupg, ~/.ssh, run sockets,/bin/gpg*,gpg-agent,ssh*,ssh-agent,pinentry etc. Profiles in question: git/frontends,password managers,email/chat clients... ?

**Comment by rusty-snake** [comment]
> I think we should make a allow-gpg/ssh.inc …

We can't. If you put `whitelist ${HOME}/.ssh` in it, you break whitelisting profiles, omitting it makes it useless. Profiles without `private-bin` don't need it and break, profiles with it need it. . . .

File: etc/profile-a-l/balsa.profile (hunk 1 above)
Comment (rusty-snake) [reply etc/profile-a-l/balsa.profile]: I mean you need /var/mail and /var/spool/mail, compare mutt or thunderbird. And if you `include whitelist-var-common.inc` you need to `whitelist` these locations under /var.

**Comment by rusty-snake** [comment]
1. /run/user/UID not ~/.gnupg

**Comment by kortewegdevries** [comment]
Do you want to switch other mail clients too on a whitelist basis? Evolution,geary...

File: etc/profile-a-l/balsa.profile (hunk 1 above)
Comment (kortewegdevries) [reply etc/profile-a-l/balsa.profile:14]: I missed this comment, I was going by balsa's config box which needs /var/mail..

**Comment by rusty-snake** [comment]
Thunderbird is whitelisting and it would be good if the others where (commented) whitelisting profiles too. (Up for a new PR).

**Comment by kortewegdevries** [comment]
> if the others where (commented) whitelisting profiles too

What do you mean by commented whitelisting?

**Comment by rusty-snake** [comment]
opt-in. IDK how much breaktage it will bring.

**Comment by kortewegdevries** [comment]
> opt-in. IDK how much breaktage it will bring.

I used geary and evolution; geary doesn't have GPG; evolution needs to be redone since it's a redirect to firefox...

**Comment by kortewegdevries** [comment]
I can push evolution here? The other ones I haven't/won't use...

**Comment by rusty-snake** [comment]
Separate one, so we can merge it independently.

File: etc/profile-a-l/kube.profile [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+nou2f
+novideo
+protocol unix,inet,inet6
+seccomp
+shell none
+tracelog
+
+# disable-mnt
+# Add "gpg,gpg2,gpg-agent,pinentry-curses,pinentry-emacs,pinentry-fltk,pinentry-gnome3,pinentry-gtk,pinentry-gtk2,pinentry-gtk-2,pinentry-qt,pinentry-qt4,pinentry-tty,pinentry-x2go,pinentry-kwallet" for gpg 
+# # Add "sh,bash,which,firefox" for hyperlinks
```
Comment (rusty-snake) [inline etc/profile-a-l/kube.profile:67]:
```suggestion
# Add "ignore private-bin" for hyperlinks or have a look at the private-bins in firefox.profile and firefox-common.profile.
```

File: etc/profile-m-z/trojita.profile [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+notv
+nou2f
+novideo
+protocol unix,inet,inet6,netlink
+seccomp
+shell none
+tracelog
+
+# disable-mnt
+# Add "sh,bash,which,firefox" for hyperlinks
```
Comment (rusty-snake) [inline etc/profile-m-z/trojita.profile:53]: same

**Review by rusty-snake — COMMENTED** [review]
Merge conflict. Please rebase or resolve it differently.

**Comment by kortewegdevries** [comment]
Is it okay now or do I rebase? For some reason too many branches messed up by ,git/config, this was tracking a totally different branch, so, I fetched/status but couldn't find a conflict.

File: etc/inc/disable-programs.inc [hunk 5]
```diff
@@ -678,7 +684,11 @@ blacklist ${HOME}/.local/share/rhythmbox
 blacklist ${HOME}/.local/share/rtv
 blacklist ${HOME}/.local/share/scribus
 blacklist ${HOME}/.local/share/signal-cli
+<<<<<<< HEAD
+blacklist ${HOME}/.local/share/sink
+=======
 blacklist ${HOME}/.local/share/smuxi
+>>>>>>> 8d77d988d73a2404dca4db81b99eec51b84698e1
```
Comment (rusty-snake) [inline etc/inc/disable-programs.inc:691]: :see_no_evil:

File: etc/profile-a-l/balsa.profile [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+nou2f
+novideo
+protocol unix,inet,inet6
+seccomp
+shell none
+tracelog
+
+# disable-mnt
+# Add "gpg,gpg2,gpg-agent,pinentry-curses,pinentry-emacs,pinentry-fltk,pinentry-gnome3,pinentry-gtk,pinentry-gtk2,pinentry-gtk-2,pinentry-qt,pinentry-qt4,pinentry-tty,pinentry-x2go,pinentry-kwallet" for gpg 
+# # Add "sh,bash,which,firefox" for hyperlinks
```
Comment (rusty-snake) [inline etc/profile-a-l/balsa.profile:62]: Forgotten to point you here.

**Review by rusty-snake — COMMENTED** [review]
Rest looks great.

**Comment by rusty-snake** [comment]
Thanks for this contribution!

**Pull request merged**